dc/s2n-quic-dc/src/uds/sender.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use nix::{
+    fcntl::{fcntl, FcntlArg, OFlag},
+    sys::socket::{
+        bind, sendmsg, socket, AddressFamily, ControlMessage, MsgFlags, SockFlag, SockType,
+        UnixAddr,
+    },
+    unistd::unlink,
+};
+use std::{
+    os::{
+        fd::{AsFd, OwnedFd},
+        unix::io::{AsRawFd, RawFd},
+    },
+    path::Path,
+};
+use tokio::io::{unix::AsyncFd, Interest};
+
+pub struct Sender {
+    socket_fd: AsyncFd<OwnedFd>,
+}
+
+impl Sender {
+    pub fn new(socket_path: &Path) -> Result<Self, std::io::Error> {
+        if socket_path.exists() {
+            unlink(socket_path)?;
+        }
+
+        let socket_owned = socket(
+            AddressFamily::Unix,
+            SockType::Datagram,
+            SockFlag::empty(),
+            None,
+        )?;
+
+        let flags = fcntl(socket_owned.as_fd(), FcntlArg::F_GETFL)?;
+        let new_flags = OFlag::from_bits_truncate(flags) | OFlag::O_NONBLOCK;
+        fcntl(socket_owned.as_fd(), FcntlArg::F_SETFL(new_flags))?;
+
+        let socket_fd = socket_owned.as_raw_fd();
+        let unix_addr = UnixAddr::new(socket_path)?;
+        bind(socket_fd, &unix_addr)?;
```

Comment:
Actually I don't think `bind` on the sender side is necessary? I can remove that